### PR TITLE
Allow period and hyphen in attribute names

### PR DESCRIPTION
**Pull request by Dave-Allured**
See issue #477 for discussion of these changes.

# Release checklist
- [x] Authors updated in `cf-conventions.adoc`?
- [ ] Next version in `cf-conventions.adoc` up to date? Versioning inspired by [SemVer](https://semver.org).
- [x] `history.adoc` up to date?
- [x] Conformance document up-to-date?

# For maintainers
After the merge remember to delete the source branch.
Tags are set at the conclusion of the annual meeting; until then, `main` always is a draft for the next version.

### CODE REVIEW

File: ch02.adoc [hunk 1]
```diff
@@ -51,6 +51,7 @@ By the word _letters_ we mean the standard ASCII letters uppercase `A` to `Z` an
 By the word _digits_ we mean the standard ASCII digits `0` to `9`, and similarly _underscores_ means the standard ASCII underscore `_`.
 Note that this is in conformance with the COARDS conventions, but is more restrictive than the netCDF interface which allows almost all Unicode characters encoded as multibyte UTF-8 characters (link:$$https://docs.unidata.ucar.edu/nug/current/file_format_specifications.html$$[NUG Appendix B]).
 The netCDF interface also allows leading underscores in names, but the NUG states that this is reserved for system use.
+ASCII period (.) and ASCII hyphen (-) are also allowed in attribute names only.
```
Comment (larsbarring) [inline ch02.adoc:55]: Perhaps backticks "`"

```ASCII period (.) and ASCII hyphen (-) ... ```

to
``` ASCII period (`.`) and ASCII hyphen (`-`) ... ```

to keep it consistent with previous text?

File: conformance.adoc [hunk 2]
```diff
@@ -37,6 +37,7 @@ See https://github.com/ugrid-conventions/ugrid-conventions for the UGRID conform
 *Recommendations:*
 
 * Variable, dimension and attribute names should begin with a letter and be composed of letters (A-Z, a-z), digits (0-9), and underscores(_). This corresponds to ASCII characters in the decimal ranges (65-90), (97-122), (48-57), and (95). The corresponding Unicode codepoints are (U+0041-U+005A), (U+0061-U+007A), (U+0030-U+0039), and (U+005F).  
+* ASCII period (.) and ASCII hyphen (-) may also be included in attribute names only.
```
Comment (larsbarring) [inline conformance.adoc:40]: Add decimal codes and Unicode codepoints as per [suggestion ](https://github.com/cf-convention/cf-conventions/issues/477#issuecomment-1965251024)

File: ch02.adoc (hunk 1 above)
Comment (Dave-Allured) [reply ch02.adoc:55]: I respectfully disagree with your requested changes.  The text is clear as I have drafted.

File: conformance.adoc (hunk 2 above)
Comment (Dave-Allured) [reply conformance.adoc:40]: I respectfully disagree with your requested changes.  The text is clear as I have drafted.

**Pull request merged**